agent/service.go

```diff
@@ -126,6 +126,10 @@ func (as *agentService) Algo(ctx context.Context, algorithm Algorithm) error {
 
 	as.algorithm = f.Name()
 
+	if len(as.computation.Datasets) == 0 {
+		as.sm.SendEvent(dataReceived)
```

Comment:
This is a hack. I would rather implement more correct approach, and that is that state machine goes directly to running if Dataset is 0, i.e. to go in data receiving stage only if there are some data to be received - otherwise go to running mode.

Reply:
Yes. It goes to `running` state. This is the event that triggers the change of state from `receivingAlgorithm` to `running`

Reply:
you could move this check into the state call back functions for  algo, such that it is declares in `New` agent function which has the manifest